TabloidMVC/Repositories/CategoryRepository.cs

```diff
@@ -95,5 +95,25 @@ OUTPUT INSERTED.ID
             }
         }
 
+        public void DeleteCategory(int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
```

Comment:
This code works perfectly if the category doesn't have any posts attached to it, but if the category contains one or more posts it throws an exception and reloads the delete form without telling the user what happened. Go find @Tommy-Spurlock and talk to him about options for deleting when you have related data! It's tricky but worth it :)